onnxscript/ir/_name_authority.py

```diff
@@ -12,20 +12,53 @@ class NameAuthority:
     ``node_{op_type}_{node_counter}`` for nodes. The counter is incremented each time
     a new value or node is named.
 
-    The class does not keep track of the names it has given, so it is possible to
-    generate names that conflicts with existing names. It is the responsibility of the
-    user to ensure that the names are unique (typically by running a name-fixing pass
-    on the graph).
+    This class does keep track of the names it has given and all the existing names
+    in the graph to prevent producing duplicated names.
+
+    However, if a value/node is already named when added to the graph,
+    the name authority will not change its name.
+    It is the responsibility of the user to ensure that the names are unique
+    (typically by running a name-fixing pass on the graph).
+    TODO(justichuby): Describe the pass when we have a reference implementation.
     """
 
     def __init__(self):
         self._value_counter = 0
         self._node_counter = 0
+        self._value_names: set[str] = set()
+        self._node_names: set[str] = set()
+
+    def _unique_value_name(self) -> str:
+        """Generate a unique name for a value."""
+        while True:
+            name = f"val_{self._value_counter}"
+            self._value_counter += 1
+            if name not in self._value_names:
+                return name
+
+    def _unique_node_name(self, op_type: str) -> str:
```

Comment:
Can this just be any prefix that the user chooses? If so, we can rename this to be `preferred_name`. Trying to separate the mechanism here from the naming-policy (which a user might want to control). `node_` could be a default-value for this  prefix.

Reply:
From discussion, users can use a pass to reconcile conflicts if they decide to specify their own names, or they can build an auxiliary data structure to keep track of names with stronger assumptions, for now. We can evolve the mechanism later when needed.